editor/config_loader.hpp

```diff
@@ -0,0 +1,62 @@
+#pragma once
+
+#include "base/exception.hpp"
+
+#include "std/chrono.hpp"
+#include "std/mutex.hpp"
+#include "std/string.hpp"
+#include "std/thread.hpp"
+
+namespace pugi
+{
+  class xml_document;
+}
+
+namespace editor
+{
+// Class for multithreaded interruptable waiting
+class Waiter
+{
+public:
+
+  template <class _Rep, class _Period>
```

Comment:
s/_Rep/Rep/
s/_Period/Period/
Why template method is not defined here? It's in the public section and may be called from everywhere.